File: src/lib/mobile.ts
import { invoke } from "@tauri-apps/api/core";

export interface MobileDevice {
  id: string;
  name: string;
  platform: string;
  last_seen: string;
  is_online: boolean;
}

export interface SyncStatus {
  last_sync: string | null;
  pending_changes: number;
  syncing: boolean;
  error: string | null;
}

export interface MobileResult {
  success: boolean;
  message: string;
}

export async function getSyncStatus(): Promise<SyncStatus> {
  return invoke<SyncStatus>("get_sync_status");
}

export async function triggerSync(): Promise<SyncStatus> {
  return invoke<SyncStatus>("trigger_sync");
}

export async function listConnectedDevices(): Promise<MobileDevice[]> {
  return invoke<MobileDevice[]>("list_connected_devices");
}

export async function getMobileLinkCode(): Promise<string> {
  return invoke<string>("get_mobile_link_code");
}

export async function registerMobileDevice(
  code: string,
  deviceName: string,
  platform: string,
): Promise<MobileResult> {
  return invoke<MobileResult>("register_mobile_device", {
    code,
    deviceName,
    platform,
  });
}

export async function disconnectMobileDevice(
  deviceId: string,
): Promise<MobileResult> {
  return invoke<MobileResult>("disconnect_mobile_device", { deviceId });
}

export async function getBrainLibVersion(): Promise<string> {
  return invoke<string>("get_brain_lib_version");
}

export function formatLastSync(lastSync: string | null): string {
  if (!lastSync) return "Never";
  const date = new Date(lastSync);
  return date.toLocaleString();
}

export function getPlatformIcon(platform: string): string {
  switch (platform.toLowerCase()) {
    case "ios":
      return "🍎";
    case "android":
      return "🤖";
    default:
      return "📱";
  }
}

File: src-tauri/src/commands/mobile.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MobileDevice {
    pub id: String,
    pub name: String,
    pub platform: String,
    pub last_seen: String,
    pub is_online: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SyncStatus {
    pub last_sync: Option<String>,
    pub pending_changes: u32,
    pub syncing: bool,
    pub error: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MobileResult {
    pub success: bool,
    pub message: String,
}

#[tauri::command]
pub fn get_sync_status() -> SyncStatus {
    SyncStatus {
        last_sync: None,
        pending_changes: 0,
        syncing: false,
        error: None,
    }
}

#[tauri::command]
pub async fn trigger_sync() -> Result<SyncStatus, String> {
    Ok(SyncStatus {
        last_sync: Some(chrono::Utc::now().to_rfc3339()),
        pending_changes: 0,
        syncing: false,
        error: None,
    })
}

#[tauri::command]
pub fn list_connected_devices() -> Vec<MobileDevice> {
    Vec::new()
}

#[tauri::command]
pub fn get_mobile_link_code() -> String {
    use uuid::Uuid;
    let code = Uuid::new_v4().to_string()[..8].to_uppercase();
    format!("TT-{}", code)
}

#[tauri::command]
pub fn register_mobile_device(code: String, device_name: String, platform: String) -> Result<MobileResult, String> {
    log::info!("Mobile device {} registered with code {}", device_name, code);
    Ok(MobileResult {
        success: true,
        message: format!("Device {} linked successfully", device_name),
    })
}

#[tauri::command]
pub fn disconnect_mobile_device(device_id: String) -> Result<MobileResult, String> {
    Ok(MobileResult {
        success: true,
        message: format!("Device {} disconnected", device_id),
    })
}

#[tauri::command]
pub fn get_brain_lib_version() -> String {
    env!("CARGO_PKG_VERSION").to_string()
}